hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/operations/AddAllOperation.java

```diff
@@ -0,0 +1,133 @@
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.ringbuffer.OverflowPolicy;
+import com.hazelcast.ringbuffer.impl.RingbufferContainer;
+import com.hazelcast.spi.BackupAwareOperation;
+import com.hazelcast.spi.Notifier;
+import com.hazelcast.spi.Operation;
+import com.hazelcast.spi.WaitNotifyKey;
+
+import java.io.IOException;
+
+import static com.hazelcast.ringbuffer.OverflowPolicy.FAIL;
+import static com.hazelcast.ringbuffer.OverflowPolicy.values;
+import static com.hazelcast.ringbuffer.impl.RingbufferDataSerializerHook.ADD_ALL_OPERATION;
+
+public class AddAllOperation extends AbstractRingBufferOperation
+        implements Notifier, BackupAwareOperation {
+
+    private OverflowPolicy overflowPolicy;
+    private Data[] items;
+    private long resultSequence;
+
+    public AddAllOperation() {
+    }
+
+    public AddAllOperation(String name, Data[] items, OverflowPolicy overflowPolicy) {
+        super(name);
+        this.items = items;
+        this.overflowPolicy = overflowPolicy;
+    }
+
+    @Override
+    public void run() throws Exception {
+        RingbufferContainer ringbuffer = getRingBufferContainer();
+
+        if (overflowPolicy == FAIL) {
+            if (ringbuffer.remainingCapacity() < items.length) {
+                resultSequence = -1;
+                return;
+            }
+        }
+
+        resultSequence = ringbuffer.addAll(items);
+    }
+
+    @Override
+    public Object getResponse() {
+        return resultSequence;
+    }
+
+    @Override
+    public boolean shouldNotify() {
+        return resultSequence != -1;
+    }
+
+    @Override
+    public WaitNotifyKey getNotifiedKey() {
+        RingbufferContainer ringbuffer = getRingBufferContainer();
+        return ringbuffer.getRingEmptyWaitNotifyKey();
+    }
+
+    @Override
+    public boolean shouldBackup() {
+        return resultSequence != -1;
+    }
+
+    @Override
+    public int getSyncBackupCount() {
+        RingbufferContainer ringbuffer = getRingBufferContainer();
+        return ringbuffer.getConfig().getBackupCount();
+    }
+
+    @Override
+    public int getAsyncBackupCount() {
+        RingbufferContainer ringbuffer = getRingBufferContainer();
+        return ringbuffer.getConfig().getAsyncBackupCount();
+    }
+
+    @Override
+    public Operation getBackupOperation() {
+        return new AddAllBackupOperation(name, items);
+    }
+
+    @Override
+    public int getId() {
+        return ADD_ALL_OPERATION;
+    }
+
+    @Override
+    protected void writeInternal(ObjectDataOutput out) throws IOException {
+        super.writeInternal(out);
+
+        out.writeInt(overflowPolicy.ordinal());
+
+        out.writeInt(items.length);
+        for (Data item : items) {
+            out.writeData(item);
+        }
+    }
+
+    @Override
+    protected void readInternal(ObjectDataInput in) throws IOException {
+        super.readInternal(in);
+
+        overflowPolicy = values()[in.readInt()];
```

Comment:
To me, `values()` static import makes this harder to read. I prefer to see `OverflowPolicy.values()`.